phantom-dsl/src/main/scala/com/outworkers/phantom/column/MapColumn.scala

```diff
@@ -61,17 +65,53 @@ private[phantom] abstract class AbstractMapColumn[
 class MapColumn[Owner <: CassandraTable[Owner, Record], Record, K : Primitive, V : Primitive](table: CassandraTable[Owner, Record])
     extends AbstractMapColumn[Owner, Record, K, V](table) with PrimitiveCollectionValue[V] {
 
-  val keyPrimitive = Primitive[K]
+  private[this] val keyPrimitive = Primitive[K]
 
   override def keyAsCql(v: K): String = keyPrimitive.asCql(v)
 
-  override val valuePrimitive = Primitive[V]
+  override val valuePrimitive: Primitive[V] = Primitive[V]
 
-  override val cassandraType = QueryBuilder.Collections.mapType(
+  override val cassandraType: String = QueryBuilder.Collections.mapType(
     keyPrimitive.cassandraType,
     valuePrimitive.cassandraType
   ).queryString
 
+  /*
```

Comment:
@iRhysBradbury This will need to replace the Java Driver collection parsing mechanism which is painfully slow, but it can go away from the code for now.